### PR TITLE
Project Crud

### PR DESCRIPTION
**Pull request by Georgeygigz**
**What does this PR do?**

 - Create implementation to create a project
 - Create implementation to get lists of project

**description of Task to be completed?**

- Create implementation to create a project

**How should this be manually tested?**

- `git checkout ft-create-project`
- endpoint `{base_url}/project`
- payload ```{
  "trackId": "5fdd8c4ad43f41c6f77bc0cf",
  "topic": "Regex",
  "duration": "2hrs",
  "projectDescription": "Regular expression validation",
	"team":""5fdd8c4ad43f41c6f77bc0cf
}```

**Any background context you want to provide?**

 - N/A

**What are the relevant Issue Number?**

[#34](https://github.com/Nicanor008/shoman-FE/issues/34)

**Screenshots (if appropriate)**

- N/A

**Questions:**

- N/A

### CODE REVIEW

File: src/api/v1/project/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express')
+import { CreateProjectController, GetProjectsController } from './project_controllers'
+import { isAuthenticated, isAdmin, isMentor } from '../../../middlewares/auth'
+
+const router = express.Router()
+
+router.post('/project', isAuthenticated, CreateProjectController)
```
Comment (neymarjimoh) [inline src/api/v1/project/index.js:7]: Any logged in can create a project?

Comment (Nicanor008) [reply src/api/v1/project/index.js:7]: should be `isMentor` as only mentors can create projects and learning content

File: src/api/v1/project/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require('express')
+import { CreateProjectController, GetProjectsController, getSingleProject } from './project_controllers'
+import { isAuthenticated, isMentor } from '../../../middlewares/auth'
+
+const router = express.Router()
+
+router.post('/project', isMentor, CreateProjectController)
+router.get('/projects', isAuthenticated, GetProjectsController)
+router.get('/project/:projectId>', isAuthenticated, getSingleProject)
```
Comment (Nicanor008) [inline src/api/v1/project/index.js:9]: This should be uniform 
```suggestion
router.get('/project', isAuthenticated, GetProjectsController)
router.get('/project/:projectId', isAuthenticated, getSingleProject)
```

**Review by Nicanor008 — APPROVED** [review]
Good job @Georgeygigz, we can enhance later, for now, it's good to go.

File: src/api/v1/project/index.js (hunk 2 above)
Comment (Nicanor008) [reply src/api/v1/project/index.js:9]: or all can be `/projects`

**Review by Nicanor008 — APPROVED** [review]

**Pull request merged**